packages/plugins/Traderxyz/src/SNSAdaptor/PostPreview.tsx

```diff
@@ -0,0 +1,301 @@
+import { Box, Chip, Grid, Typography, useTheme } from '@mui/material'
+import { makeStyles, useCustomSnackbar } from '@masknet/theme'
+import { useI18N } from '../locales/i18n_generated'
+import { usePluginWrapper } from '@masknet/plugin-infra/content-script'
+import { useAccount, useChainId } from '@masknet/web3-shared-evm'
+import { useTraderApi } from '../apis/nftSwap'
+import type { TradeMetaData, NFTData } from '../types'
+import type { SwappableAsset } from '@traderxyz/nft-swap-sdk'
+import { ActionButtonPromise } from '../../../../mask/src/extension/options-page/DashboardComponents/ActionButton'
```

Comment:
You should not import from `mask`.

Reply:
It is available in shared?

Reply:
Maybe you should avoid using this component. ActionButtonPromise is overcomplicated and we want to get rid of it.